modules/table/src/test/java/org/apache/ignite/internal/table/distributed/AbstractMvStorageUpdateHandlerTest.java

```diff
@@ -0,0 +1,144 @@
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import java.util.Map;
+import java.util.UUID;
+import org.apache.ignite.distributed.TestPartitionDataStorage;
+import org.apache.ignite.internal.configuration.testframework.ConfigurationExtension;
+import org.apache.ignite.internal.configuration.testframework.InjectConfiguration;
+import org.apache.ignite.internal.distributionzones.configuration.DistributionZoneConfiguration;
+import org.apache.ignite.internal.hlc.HybridTimestamp;
+import org.apache.ignite.internal.schema.BinaryRow;
+import org.apache.ignite.internal.schema.configuration.TableConfiguration;
+import org.apache.ignite.internal.schema.configuration.TablesConfiguration;
+import org.apache.ignite.internal.storage.BaseMvStoragesTest;
+import org.apache.ignite.internal.storage.MvPartitionStorage;
+import org.apache.ignite.internal.storage.RowId;
+import org.apache.ignite.internal.storage.engine.MvTableStorage;
+import org.apache.ignite.internal.storage.engine.StorageEngine;
+import org.apache.ignite.internal.table.distributed.raft.PartitionDataStorage;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+/**
+ * Abstract class for testing {@link StorageUpdateHandler} using different implementations of {@link MvPartitionStorage}.
+ */
+@ExtendWith(ConfigurationExtension.class)
+abstract class AbstractMvStorageUpdateHandlerTest extends BaseMvStoragesTest {
+    private static final int PARTITION_ID = 0;
+
+    private StorageEngine storageEngine;
+
+    private TestPartitionDataStorage partitionDataStorage;
+
+    private StorageUpdateHandler storageUpdateHandler;
+
+    @BeforeEach
+    void setUp(
+            @InjectConfiguration("mock.tables.foo{}") TablesConfiguration tablesConfig,
+            @InjectConfiguration DistributionZoneConfiguration distributionZoneConfig
+    ) {
+        storageEngine = createStorageEngine();
+
+        storageEngine.start();
+
+        TableConfiguration tableConfig = tablesConfig.tables().get("foo");
+
+        assertThat(
+                tableConfig.dataStorage().change(dataStorageChange -> dataStorageChange.convert(storageEngine.name())),
+                willCompleteSuccessfully()
+        );
+
+        MvTableStorage mvTableStorage = storageEngine.createMvTable(tableConfig, tablesConfig, distributionZoneConfig);
+
+        mvTableStorage.start();
+
+        MvPartitionStorage mvPartitionStorage = getOrCreateMvPartition(mvTableStorage, PARTITION_ID);
+
+        partitionDataStorage = new TestPartitionDataStorage(mvPartitionStorage);
+
+        storageUpdateHandler = new StorageUpdateHandler(PARTITION_ID, partitionDataStorage, Map::of, tableConfig.dataStorage());
+    }
+
+    @AfterEach
+    void tearDown() {
+        if (storageEngine != null) {
+            storageEngine.stop();
+        }
```

Comment:
You forgot to stop the table, and partition

Reply:
Fix it